### DISCUSSION TITLE
How to pass `sslmode: prefer` to postgres?

### DISCUSSION THREAD

**Discussion by hardillb** [sequelize/sequelize#16287] [Help & Questions]
I'm trying to allow optional SSL support when connecting to a PostgreSQL db and I can't seam to get it to work.

Looking at the doc* it implies that the following should work

```
const {Sequelize} = require('sequelize')

const dbOptions = {
        dialect: 'postgres',
        host: 'xxxxxx.8nj.cockroachlabs.cloud',
        port: 26257,
        username: 'ben',
        password: 'xxxxxxx',
        database: 'foo',
        dialectOptions: {
                ssl: {
                        sslmode: 'prefer'
                }
        }
}

const sequelize = new Sequelize(dbOptions)

sequelize.authenticate()
.then(() => {
  console.log("connected")
})
.catch((err) => {
  console.log(err)
})

```

But that appears to enforce SSL even if the database doesn't support it and behaves the same as setting `dielectOptions.ssl = true`

Is there a way to pass configuration as an object and get this to work or do I need to turn the `dbOptions` object into a URL and add `?sslmode=prefer` to the end?

* https://sequelize.org/docs/v6/other-topics/dialect-specific-things/#postgresql
* https://node-postgres.com/features/ssl
* https://www.postgresql.org/docs/11/libpq-connect.html#id-1.7.3.8.3.6

**Comment by WikiRik** [comment]
A quick look at the pg-connection-string tests seems to suggest that `sslmode: prefer` is parsed the same as `ssl = true`; https://github.com/brianc/node-postgres/blob/master/packages/pg-connection-string/test/parse.js#L259

Can you check how the dialectOptions is sent to the pg connector? I'm not sure if this is a sequelize issue

**Comment by hardillb** [comment]
The sequlize [docs](https://sequelize.org/docs/v6/other-topics/dialect-specific-things/#postgresql) say the `ssl` object is passed as is directly to the pg module and the pg [docs](https://node-postgres.com/features/ssl#usage-with-connectionstring) say as long as you don't add sslmode (or similar) to the URL then it will use what's in the ssl object

Unless I'm reading the doc wrong?

Also just to be clear, I'm not necessarily  suggesting sequelize is broken, I'm more wanting to know how to get optional SSL to work.

**Comment by hardillb** [comment]
OK, looks like it's a problem with pg connector

https://github.com/brianc/node-postgres/issues/2720
